lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Shared.hs

```diff
@@ -641,16 +643,36 @@ instance SupportsDiscovery n k => CompareDiscovery (SharedState n k) where
                     AddressPool.lookup addr (getPool intPool)
 
 instance Typeable n => KnownAddresses (SharedState n k) where
-    knownAddresses st = nonChangeAddresses
+    knownAddresses st =
+        nonChangeAddresses <> usedChangeAddresses <> pendingChangeAddresses
```

Comment:
This code now matches the [corresponding logic for `Seq` states](https://github.com/input-output-hk/cardano-wallet/blob/2863d20c83175a262c0712a9d02ee195ac8feed6/lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs#L662-L695) more closely.

Question: What about the order in which the addresses are returned? 🤔 The other code applies `L.sortOn idx` in order to sort the addresses from lowest to highest index. Do we expect `knownAddresses` to return addresses in a specific order?

Reply:
indeed, we should conserve ordering as we present addresses! Done